web_widget_url_advanced/static/src/fields/url_field.esm.js

```diff
@@ -0,0 +1,56 @@
+/** @odoo-module **/
+
+/* Copyright 2018 Simone Orsi - Camptocamp SA
+License LGPLv3.0 or later (https://www.gnu.org/licenses/lgpl-3.0.en.html). */
+
+import {patch} from "@web/core/utils/patch";
+import {UrlField, formUrlField, urlField} from "@web/views/fields/url/url_field";
+
+patch(UrlField.props, {
+    text_field: {type: String, optional: true},
+    prefix_name: {type: String, optional: true},
+});
+
+patch(UrlField.prototype, {
+    _get_text_field() {
+        if (this.props.text_field) {
+            let field_value = this.props.record.data[this.props.text_field];
+            if (typeof field_value === "object" && field_value.length == 2) {
```

Comment:
```suggestion
            if (Array.isArray(field_value) && field_value.length == 2) {
```

seems to narrow the conditions